Backbone.Tests.ArchUnit/DomainEvents.cs

```diff
@@ -0,0 +1,57 @@
+﻿using ArchUnitNET.Domain.Extensions;
+using ArchUnitNET.Fluent.Conditions;
+using ArchUnitNET.xUnit;
+using Backbone.BuildingBlocks.Domain.Events;
+using Backbone.Tooling.Extensions;
+using static ArchUnitNET.Fluent.ArchRuleDefinition;
+
+namespace Backbone.Backbone.Tests.ArchUnit;
+
+public class DomainEvents
+{
+    [Fact]
+    public void IncomingDomainEventsShouldNotHaveConstructors()
+    {
+        Types()
+            .That().AreAssignableTo(typeof(DomainEvent))
+            .And().ResideInNamespace("Backbone.Modules.*.Domain.DomainEvents.Incoming.*", true)
+            .Should().FollowCustomCondition(type =>
+            {
+                if (type.GetConstructors().Any(c => c.Parameters.Any()))
+                    return new ConditionResult(type, false, "should not have constructors");
+
+                return new ConditionResult(type, true);
+            }, "")
+            .Because("otherwise the JSON Serializer would call the constructor instead of serializing directly, which would cause an invalid serialization")
+            .Check(Backbone.ARCHITECTURE);
+    }
+
+    [Fact]
+    public void OutgoingDomainEventsShouldHaveCorrespondingIncomingDomainEvents()
+    {
+        var outgoing = Types()
+            .That().AreAssignableTo(typeof(DomainEvent))
+            .And().ResideInNamespace("Backbone.Modules.*.DomainEvents.Outgoing.*", true)
+            .GetObjects(Backbone.ARCHITECTURE)
+            .ToList();
+
+        Types()
+            .That().AreAssignableTo(typeof(DomainEvent))
+            .And().ResideInNamespace("Backbone.Modules.*.DomainEvents.Incoming.*", true)
```

Comment:
Since you're using the incoming domain events multiple times, it probably makes sense to introduce a field for them.
And for the sake of consistency, you could do the same for the outgoing ones